library/src/main/java/com/okta/oidc/clients/web/WebAuthClientImpl.java

```diff
@@ -130,10 +132,23 @@ public boolean isInProgress() {
     public void signIn(@NonNull final Activity activity, AuthenticationPayload payload) {
         registerActivityLifeCycle(activity);
         cancelFuture();
+        //add the login hint if it exists.
+        if (payload == null && mLoginHint != null) {
+            payload = new AuthenticationPayload.Builder().setLoginHint(mLoginHint).build();
+        } else if (mLoginHint != null) {
+            Map<String, String> params = payload.getAdditionalParameters();
+            AuthenticationPayload.Builder builder = new AuthenticationPayload.Builder();
+            for (Map.Entry<String, String> param : params.entrySet()) {
+                builder.addParameter(param.getKey(), param.getValue());
+            }
+            builder.setLoginHint(mLoginHint);
```

Comment:
we could break this out to its own if statement for clarity with the builder being initialized before either if group.

Reply:
I'll add a copy method in the builder and receives a AuthenticationPayload as a parameter.